### ISSUE TITLE
Introduce the "in" relational operator

### ISSUE THREAD

**Issue by raducotescu** [Adobe-Marketing-Cloud/htl-spec#56]
The `in` relational operator can be used to:
1. Check if a `String` is contained by another `String`:
```html
${'a' in 'abc'} <!--/* returns true */-->
${'ab' in 'abc'} <!--/* returns true */-->
${'bc' in 'abc'} <!--/* returns true */-->
${'abc' in 'abc'} <!--/* returns true */-->
${'d' in 'abc'} <!--/* returns false */-->
```

2. Check if an `Array` or a `List` contains an object:
```html
<!--/*
  Assuming myArray would be in scope and would have the following content:
  [100, 200, 300, 400, 500]
*/-->
${100 in myArray} <!--/* returns true */-->
${300 in myArray} <!--/* returns true */-->
${1 in myArray} <!--/* returns false */-->
```

3. Check if an object has a property or a `Map` has a key:

    Assuming the following use object (more details in the Use-API section) provided by a `logic.js` file:
    ```javascript
    use(function () {
        return {
            a: true,
            b: 'two',
            c: 3
        };
    });
    ```

    ```html
    <sly data-sly-use.logic="logic.js" />
    ${'a' in logic} <!--/* returns true */-->
    ${'b' in logic} <!--/* returns true */-->
    ${'c' in logic} <!--/* returns true */-->
    ${'two' in logic} <!--/* returns false */-->
    ```

**Comment by raducotescu** [comment]
Added to spec in 611c11b3d59aa0a12633cc858af19e8154b994ab.

**Comment by karollewandowski** [comment]
*I don't want to open new issue with question related to this one.*

How grammar will be extended to cover `in` operator?

**Comment by raducotescu** [comment]
I will add it as soon as I also fix #53, which is basically an issue about operators priority.
